GVFS/GVFS.Common/FileSystem/IPlatformFileSystem.cs

```diff
@@ -8,5 +8,42 @@ public interface IPlatformFileSystem
         void CreateHardLink(string newLinkFileName, string existingFileName);
         bool TryGetNormalizedPath(string path, out string normalizedPath, out string errorMessage);
         void ChangeMode(string path, int mode);
+
+        /// <summary>
+        /// Check if a path exists as a subpath of the specified directory.
+        /// </summary>
+        /// <param name="directoryPath">Directory path.</param>
+        /// <param name="path">Path to query.</param>
+        /// <returns>True if <see cref="path"/> exists as a subpath of <see cref="directoryPath"/>, false otherwise.</returns>
+        bool IsPathUnderDirectory(string directoryPath, string path);

```

Comment:
Why does this method exist here in this interface? Isn't this a platform-agnostic question?

Reply:
Perhaps, although case sensitivity when comparing path segments is file system (and transitively OS) dependent?

Reply:
I see, that is a valid concern. However, note that we are currently very unprepared to deal with case sensitive file systems. Note that even putting it here in this interface doesn't solve that problem, since it's not a per-OS question but rather a per-file-system-and-machine-configuration question. We will have to deal with this as a broader issue since we are now porting to Linux as well, but it is outside the scope of this PR.